src/Adapter/Requests/Services/DefaultBucketService.cs

```diff
@@ -46,17 +56,81 @@ public async Task<Bucket> GetBucketAndWaitForAvailability(Request request, Cance
-            await repository.Save(bucket, cancellationToken);
             return bucket;
         }
 
+        /// <inheritdoc />
+        public async Task<Bucket> MergeBucketIds(Bucket bucket, HttpResponseMessage response, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            if (!response.Headers.TryGetValues(RateLimitBucketHeader, out var headerValues))
+            {
+                logger.LogWarning($"Response did not have {RateLimitBucketHeader} header.");
+                return bucket;
+            }
+
+            var remoteId = headerValues.First();
+            if (bucket.RemoteId != null)
+            {
+                bucket.RemoteId = remoteId;
+                return bucket;
+            }
+
+            var existingBucket = await repository.FindByRemoteId(remoteId, cancellationToken);
+            if (existingBucket == null)
+            {
+                bucket.RemoteId = remoteId;
+                return bucket;
+            }
+
+            repository.Remove(bucket);
+            existingBucket.Endpoints |= bucket.Endpoints;
+            return existingBucket;
+        }
+
+        /// <inheritdoc />
+        public void UpdateBucketResetAfter(Bucket bucket, HttpResponseMessage response)
+        {
+            if (!response.Headers.TryGetValues(RateLimitResetHeader, out var values))
+            {
+                logger.LogWarning($"Response did not have {RateLimitResetHeader} header.");
+                return;
+            }
+
+            if (!long.TryParse(values.First(), out var resetAfterSeconds))
+            {
+                logger.LogWarning($"Response had a non-numeric {RateLimitResetHeader} header value.");
+                return;
+            }
+
+            bucket.ResetAfter = DateTimeOffset.FromUnixTimeSeconds(resetAfterSeconds);
+        }
+
+        /// <inheritdoc />
+        public void UpdateBucketHitsRemaining(Bucket bucket, HttpResponseMessage response)
+        {
+            if (!response.Headers.TryGetValues(RateLimitRemainingHeader, out var headerValues))
+            {
+                logger.LogWarning($"Response did not have {RateLimitRemainingHeader} header.");
+                return;
+            }
+
+            if (!int.TryParse(headerValues.First(), out var hitsRemaining))
+            {
+                logger.LogWarning($"Response had a non-numeric {hitsRemaining} header value.");
+                return;
+            }
+
+            bucket.HitsRemaining = hitsRemaining;
+        }
+
         private async Task<Bucket> CreateBucket(Request request, CancellationToken cancellationToken = default)
         {
             var bucket = new Bucket
             {
                 MajorParameters = request.Parameters,
                 HitsRemaining = 1,
-                ResetAfter = DateTimeOffset.Now + TimeSpan.FromMinutes(1),
+                ResetAfter = DateTimeOffset.Now + TimeSpan.FromMinutes(0.1),
```

Comment:
Could use just DateTimeOffset.Now here, already have a jitter and min cooldown of 100ms